### PR TITLE
CAMEL-16080: Migrate camel-kamelet component from camel-k-runtime to camel

### PR DESCRIPTION
**Pull request by lburgazzoli**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+====
+To let the *Kamelet* component wiring the materialized route to the caller processor, we need to be able to identify the input and output endpoint of the route and this is done by using `kamele:source` to mark the input endpoint and `kamelet:sink` for the output endpoint.
+====
+
+Then the template can be instantiated and invoked as shown below:
+
+[source,java]
+----
+from("direct:template")
+    .to("kamelet:setBody?bodyValue=kamelet");
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc:115]: Can we use mykamelet as bodyValue to make it more obvious that kamelet is not some special value

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+[IMPORTANT]
+====
+To let the *Kamelet* component wiring the materialized route to the caller processor, we need to be able to identify the input and output endpoint of the route and this is done by using `kamele:source` to mark the input endpoint and `kamelet:sink` for the output endpoint.
+====
+
+Then the template can be instantiated and invoked as shown below:
+
+[source,java]
+----
+from("direct:template")
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc:114]: myTemplate

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+== Samples
+
+
+Kamelets can be used as if they were standard Camel components.
+For example, suppose that we have created a Route Template as follows:
+
+[source,java]
+----
+routeTemplate("setBody")
```
Comment (davsclaus) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/docs/kamelet-component.adoc:99]: setMyBody

File: components/camel-kamelet/pom.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    <artifactId>camel-kamelet</artifactId>
+    <packaging>jar</packaging>
+
+    <name>Camel :: Kamelet</name>
+    <description>The Kamelet Component provides support for interacting with the Camel Route Template engine</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
```
Comment (davsclaus) [inline components/camel-kamelet/pom.xml:37]: camel-api and camel-util are transparet from camel-support

**Review by davsclaus — APPROVED** [review]
I can only find some bits in the doc page. I tend to like that things are named myXXX when its something that the end user can freely choose.

**Comment by davsclaus** [comment]
And it would be nice with an example at camel-examples git repo

**Comment by lburgazzoli** [comment]
> I can only find some bits in the doc page. I tend to like that things are named myXXX when its something that the end user can freely choose.

fixed

**Comment by lburgazzoli** [comment]
> And it would be nice with an example at camel-examples git repo

will create one once merged

**Review by oscerd — APPROVED** [review]
LGTM, I think we should merge.

**Pull request merged**